server/src/word-list.ts

```diff
@@ -0,0 +1,127 @@
+import {
+    TextDocument,
+    CompletionItem,
+    CompletionItemKind,
+    TextDocumentChangeEvent
+} from 'vscode-languageserver';
+
+// A data structure to keep track of the words used in this file
+// We currently use a set, can be changed to a trie for more efficient lookup
+class WordListClass {
+
+    // The keywords 
+    keywords : Set<string>;
+
+    // A mapping from files to the words in that file
+    dictionary: Map<TextDocument, Set<string>>;
+
+    /**
+     * Initialize the list to only contain the keywords
+     *
+     * @param {string[]} word
+     * 
+     */
+
+    constructor(word: string[]) {
+        this.keywords = new Set();
+        for (let w of word) {
+          this.keywords.add(w);
+        }
+        this.dictionary = new Map();
+    }
+
+    /**
+     * Add word to the autocomplete list for that particular document
+     *
+     * @param {string} word
+     * @param {TextDocument} d
+     * 
+     */
+    addWord(word: string, d: TextDocument) {
+
+        if (!this.dictionary.has(d)) {
+          this.dictionary.set(d,new Set());
+        }
+        if (this.dictionary.has(d)) {
```

Comment:
At this point the dictionary should already have 'd' in it, no need to check agaiin